docs/src/proposals/epoch_accounts_hash.md

```diff
@@ -0,0 +1,203 @@
+   validator will need to keep around all the state for the accounts at the
+   `start slot` while the EAH calculation is occurring; this is expensive, and
+   that duration should be minimized.
+4. The `start offset` should be *after* the `rewarding interval`
+   (from [Partitioned Inflationary Rewards Distribution](https://github.com/solana-labs/solana/pull/27455)).
+
+Once the EAH calculation is complete, it must be saved somewhere.  Since this
+occurs in the background, there is not an associated `Bank` that would make
+sense to save into.  Instead, a new field will be added to `AccountsDb` that
+will store the EAH.  Later, the bank at slot `stop slot` will read the EAH from
+`AccountsDb` and hash it into its own hash (aka _bank hash_).
+
+EAH calculation will use the existing _accounts background services_ (_ABS_) to
+perform the actual calculation.  Requests for EAH calculation will be sent from
+`bank_forks::set_root()`, with a new request type to distinguish an EAH request
+from a Snapshot request.  Since the EAH will be part of consensus, it is not
+optional; EAH requests will have the highest priority in ABS, and will be
+processed first/instead of other requests.
+
+
+#### Details
+
+#### Snapshots
+
+A snapshot contains all the state necessarily to reconstruct the cluster as of
+a certain slot.  A snapshot taken early in the epoch will need to contain the
+EAH so that the `stop slot` can include it.  Consider the following scenarios
+within an epoch where a snapshot is requested for slot `X`:
+
+
+##### 1. `X >= first slot in epoch` and `X < start slot`
+
+Since the `start slot` has not been reached yet, there is nothing special to do
+in order to take a snapshot in this scenario.
+
+
+##### 2. `X == start slot`
+
+The EAH *must* be included in the snapshot.  Since the snapshot process always
+calculates the accounts hash, no additional calculations are required.  The
+accounts hash calculation result will be used both to store in the snapshot as
+the EAH, and for the snapshot hash.
+
+
+##### 3. `X > start slot` and `X < stop slot`
+
+If a snapshot is requested *after* the `start slot` but *before* the EAH
+calculation has completed, then it will be impossible to create a snapshot with
+the correct EAH.  The snapshot process will wait until the EAH calculation has
+completed before proceeding.
+
+
+##### 4. `X == stop slot`
+
+The EAH has been calculated for this epoch, and will be included in the
+snapshot.  No further handling is required.
+
+
+##### 5. `X > stop slot` and `X <= last slot in epoch`
+
+Same as (4); the EAH has been calculated for this epoch, and will be included
+in the snapshot.  No further handling is required.
+
+
+#### Ledger-Tool
+
+Ledger-tool is used to create snapshots, so it must also be made aware of EAH.
+If a new snapshot is to be created at slot `X`, where `X > start slot` and `X <
+stop slot`, then an EAH calculation *must* occur.  It will be necessary that
```

Comment:
'EAH calculation *must* occur' OR the EAH must be in the snapshot ledger-tool loads from.